### ISSUE TITLE
I hope for auto add required to <FormItem/>

### ISSUE THREAD

**Issue by jeessy2** [jannikbuschke/formik-antd#50]
```
const SignupSchema = Yup.object().shape({
    realname: Yup.string().trim().required('realname'),
});
```
```
<FormItem name={"realname"} label={'realname'} required>
    <Input name="realname" />
</FormItem>
```

I hope for auto add required for FormItem

**Comment by jannikbuschke** [comment]
Can you provide your full source code (with the Formik container)?

Generally how validation rules are applied is not part of this library. How to add validation to formik is explained in the docs there https://jaredpalmer.com/formik/docs/guides/validation.

The (formik-antd) FormItem as you outlined above should render validation errors if formik correctly detected validation errors for the field realname (so pls provide more context).

**Comment by jeessy2** [comment]
@jannikbuschke Sorry, The validation can be working. It's not validation question!  
When I use "Yup" to validation, like this:
```
const SignupSchema = Yup.object().shape({
    realname: Yup.string().trim().required('realname'),
});
```
But, I must manually writing, like this:
```
<FormItem name={"realname"} label={'realname'} required>
    <Input name="realname" />
</FormItem>

```

So, I hope for auto add required for <FormItem/>

**Comment by jannikbuschke** [comment]
Ah ok, now I understand what you mean.

This is technically very difficult/impossible (the FormItem component does not have access to the validation rules that are configured). And formik does not specify Yup as the validation library, so only after the validation has run (which could also send a request to a server) we really know what fields are required.

**Comment by tobiasriemenschneider** [comment]
If a field is defined as being required within the YUP validation schema, this shows up in the validation schema object like this:

```js

_exclusive: {
   required: true
}

```
So I created a helper function that checks for this and conditionally returns `true`, if the field was defined to be `required`.

```js

const setRequired = (fieldName) => {
  if (AddSchema.fields[fieldName] !== undefined) {
    return AddSchema.fields[fieldName]['_exclusive']['required'];
  }
}

```

Now I can pass this function into the `required` prop of the `<Field.Item>`, passing in the `name` of the respective field.

```js

<Form.Item
  required={setRequired('name')}      //   👈👈👈
  label="Name"
  help={touched.name && errors.name ? errors.name : undefined}
  validateStatus={
    touched.name && errors.name ? 'error' : undefined
  }
>
  <Input
    name="name"
    value={values.name}
    placeholder="Vor- und Nachname"
    onChange={handleChange}
    onBlur={handleBlur}
  />
</Form.Item>

```

Not sure if this is really much more efficient than hardcoding the `required` prop on each field, though. 😅
